src/utils/timeout.ts

```diff
@@ -1,3 +1,18 @@
 export async function timeout(ms: number): Promise<void> {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
+
+export class TimeoutError extends Error {}
+
+export const promiseTimeout = async <T>(ms: number, promise: Promise<T>): Promise<T> => {
```

Comment:
please rename to something more meaningful such as "resolveWithTimeout" or "racePromiseWithTimeout"
ms -> msTimout